script.js

```diff
@@ -1,12 +1,602 @@
+  castListDiv.style.display = 'flex';
+  let castListDisplay = castList.map((cast) => {
+        let characterName = cast && cast.character ? cast.character.name : null;
+        let personName = cast.person ? cast.person.name : null;
+        let personImg = cast.person && cast.person.image ? cast.person.image.medium : null;
+
+        if (personImg === null) {
+          personImg = "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU";
+        }
+        return `
+        <div class="card cast-card mb-3">
+          <div class="row">
+            <div class="col-md-4 cast-img-div">
+              <img src="${personImg}" class="card-img-top " alt="...">
+            </div>
+            <div class="col-md-8">
+              <div class="card-body p-0">
+                <p class="character-name">${characterName}</p>
+                <p class="person-name">${personName}</p>
+              </div>
+            </div>
+          </div>
+        </div>
+        `
+  }).join('');
+
+  castListDiv.innerHTML = castListDisplay;
+};
+
+let freeSearchInput = document.getElementById('freeSearchInput');
+const freeSearchCont = document.getElementById('freeSearchCont');
+const allShowsContainer = document.getElementById('allShowsContainer');
+let showContainer = document.getElementById('showContainer');
+const castListSection = document.getElementById('castListSection');
+let seasonInputContainer = document.getElementById('seasonInputContainer');
+
+freeSearchInput.addEventListener('keyup', (e) => {
+  const freeSearchResult = document.getElementById('freeSearchResult');
+  const freeSearchResultShowsElem = document.getElementById('freeSearchResultShows');
+  let freeSearchValue = e.target.value.toLowerCase();
+  const filteredShows = allShows.filter(show => { 
+    let showName = show.name.toLowerCase();
+    let showGenres = show.genres.join(" ").toLowerCase();
+    let showSummary = show.summary.toLowerCase();
+
+    if (showName.includes(freeSearchValue) || showGenres.includes(freeSearchValue) || showSummary.includes(freeSearchValue)) {
+      return show;
+    };
+  });
+
+  console.log(filteredShows);
+  if (freeSearchValue == ""){
+    freeSearchResult.style.display = 'none';
+  } else if (filteredShows == "") {
+    freeSearchResult.style.display = 'flex';
+    freeSearchResultShowsElem.innerHTML = `<p>No shows found </br></p>`;
+  } else {
+    freeSearchResultShows(filteredShows);
+  }
+  
+  return freeSearchValue;
+});
+
+function freeSearchResultShows(shows){
+  const freeSearchResult = document.getElementById('freeSearchResult');
+  freeSearchResult.style.display = 'flex';
+  const freeSearchResultShows = document.getElementById('freeSearchResultShows');
+
+  let freeSearchResultShowsHtml = shows.map((show) => { 
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
```

Comment:
This is good, as we discussed on our call. If there's a value, use that. Otherwise, provide a placeholder.

Reply:
yeah thanks for that